pkg/storegateway/bucket.go

```diff
@@ -634,7 +634,7 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storegatewaypb.Stor
 			level.Debug(s.logger).Log("msg", "queried non-compacted block", "blockId", b.meta.ULID, "ooo", b.meta.Compaction.FromOutOfOrder())
 		}
 
-		b.queried.Store(true)
+		b.queried.CompareAndSwap(false, true)
```

Comment:
This change looks unnecessary.

Reply:
Yes, it's unrelated housekeeping I noticed, and added in a separate commit. Shall I revert?

Reply:
No worries - reverted